tests/test_ramp_fitting_likly_fit.py

```diff
@@ -0,0 +1,729 @@
+    noise with lambda 5.0.  At group 4 is a jump of 1100.0.
+    """
+    nints, ngroups, nrows, ncols = 1, 10, 1, 1
+    rnval, gval = 10.0, 5.0
+    frame_time, nframes, groupgap = 10.736, 5, 2
+
+    dims = nints, ngroups, nrows, ncols
+    var = rnval, gval
+    tm = frame_time, nframes, groupgap
+
+    ramp_data, gain2d, rnoise2d = create_blank_ramp_data(dims, var, tm)
+
+    # A randomly generated ramp by setting up a ramp that has a slope of 150.
+    # with some randomly added Poisson values, with lambda=5., and a jump
+    # at group 4.
+    ramp = np.array([153., 307., 457., 604., 1853., 2002., 2159., 2308., 2459., 2601.])
+    ramp_data.data[0, :, 0, 0] = ramp
+
+    # Create a jump, but don't mark it to make sure it gets detected.
+    dq = np.array([GOOD] * ngroups)
+    ramp_data.groupdq[0, :, 0, 0] = dq
+
+    save_opt, algo, ncores = False, "LIKELY", "none"
+    slopes, cube, ols_opt, gls_opt = ramp_fit_data(
+        ramp_data, 512, save_opt, rnoise2d, gain2d, algo, "optimal", ncores, test_dq_flags
+    )
+
+    data, dq, vp, vr, err = slopes
+    tol = 1.e-4
+
+    assert abs(data[0, 0] - 1.9960526) < tol
+    assert dq[0, 0] == JMP
+    assert abs(vp[0, 0] - 0.00064461) < tol
+    assert abs(vr[0, 0] - 0.00018037) < tol
+
+
+def test_long_ramp():
+    """
+    Test a long ramp with hundreds of groups.
+    """
+    nints, ngroups, nrows, ncols = 1, 200, 1, 1
+    rnval, gval = 10.0, 5.0
+    frame_time, nframes, groupgap = 10.736, 4, 1
+
+    dims = nints, ngroups, nrows, ncols
+    var = rnval, gval
+    tm = frame_time, nframes, groupgap
+
+    ramp_data, gain2d, rnoise2d = create_blank_ramp_data(dims, var, tm)
+
+    # Create a simple linear ramp.
+    ramp = np.array(list(range(ngroups))) * 20 + 10
+    ramp_data.data[0, :, 0, 0] = ramp
+
+    save_opt, algo, ncores = False, "LIKELY", "none"
+    slopes, cube, ols_opt, gls_opt = ramp_fit_data(
+        ramp_data, 512, save_opt, rnoise2d, gain2d, algo, "optimal", ncores, test_dq_flags
+    )
+
+    data = cube[0][0, 0, 0]
+    ddiff = (ramp_data.data[0, ngroups-1, 0, 0] - ramp_data.data[0, 0, 0, 0])
+    check = ddiff / float(ngroups-1)
+    check = check / ramp_data.group_time
+    tol = 1.e-5
+    diff = abs(data - check)
+    assert diff < tol
+
+    # Check against OLS.
+    ramp_data1, gain2d1, rnoise2d1 = create_blank_ramp_data(dims, var, tm)
+
+    ramp = np.array(list(range(ngroups))) * 20 + 10
+    ramp_data1.data[0, :, 0, 0] = ramp
+
+    save_opt, algo, ncores = False, "OLS", "none"
+    slopes1, cube1, ols_opt1, gls_opt1 = ramp_fit_data(
+        ramp_data1, 512, save_opt, rnoise2d1, gain2d1, algo, "optimal", ncores, test_dq_flags
+    )
+
+    data1 = cube1[0][0, 0, 0]
+    diff = abs(data - data1)
+    assert diff < tol
+    dbg_print_slope_slope1(slopes, slopes1, (0, 0))
+
+
+@pytest.mark.parametrize("ngroups", [1, 2])
+def test_too_few_group_ramp(ngroups):
+    """
+    It's supposed to fail.  The likelihood algorithm needs at least two
+    groups to work.
```

Comment:
Update comment/test? I think it needs 4 groups

Reply:
Likelihood-based ramp fit should work for two groups, but jump detection will not.